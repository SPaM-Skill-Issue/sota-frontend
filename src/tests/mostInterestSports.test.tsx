import { render, screen, waitFor } from "@testing-library/react";
import MostInterestSports from "../components/mostInterestSports";
import "@testing-library/jest-dom";
import { MemoryRouter } from "react-router-dom";
import {
  mockAudienceData,
  topFiveSportNames,
} from "./mocks/mockAudienceData";

describe('MostInterestSports Component Tests', () => {
  beforeEach(() => {
    global.fetch = jest.fn(() =>
      Promise.resolve({
        json: () => Promise.resolve(mockAudienceData),
      })
    ) as jest.Mock;
  });

  afterEach(() => {
    jest.restoreAllMocks();
  });

  test("renders MostInterestSports component", async () => {
    render(
      <MemoryRouter>
        <MostInterestSports />
      </MemoryRouter>
    );
    await waitFor(() => {
      const element = screen.getByText(/Top 5 sports with the most interest/i);
      expect(element).toBeInTheDocument();
    });
  });

  test("displays top five sport correctly", async () => {
    render(
      <MemoryRouter>
        <MostInterestSports />
      </MemoryRouter>
    );

    await waitFor(() => {
      // Check if each sport name from the topFiveSportNames is in the document.
      topFiveSportNames.forEach((sportName) => {
        expect(screen.getByText(sportName)).toBeInTheDocument();
      });
    });
  });

  test("Navigate to Overall audience page works correctly", async () => {
    render(
      <MemoryRouter>
        <MostInterestSports />
      </MemoryRouter>
    );

    await waitFor(() => {
      const element = screen.getByText(/Overall audience page/i);
      expect(element).toBeInTheDocument();
      expect(element).toHaveAttribute("href", "/audience");
    });
  });
});
