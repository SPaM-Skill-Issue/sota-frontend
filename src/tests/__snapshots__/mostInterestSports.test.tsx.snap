// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MostInterestSports Component Tests should renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex-row bg-belft-blue rounded-3xl"
      >
        <div
          class="flex-row w-full p-5"
        >
          <div
            class=" flex justify-end"
          >
            <span
              class="font-primary text-hunyadi-yellow text-3xl"
            >
              Top 5 sports with the most interest
            </span>
          </div>
          <div
            class="flex justify-evenly w-full py-5"
          />
          <div
            class="flex w-full justify-end pb-5 pr-5 text-white"
          >
            <a
              href="/audience"
            >
              Overall audience page 
              <span
                aria-label="arrow-right"
                class="anticon anticon-arrow-right"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="arrow-right"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M869 487.8L491.2 159.9c-2.9-2.5-6.6-3.9-10.5-3.9h-88.5c-7.4 0-10.8 9.2-5.2 14l350.2 304H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h585.1L386.9 854c-5.6 4.9-2.2 14 5.2 14h91.5c1.9 0 3.8-.7 5.2-2L869 536.2a32.07 32.07 0 000-48.4z"
                  />
                </svg>
              </span>
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex-row bg-belft-blue rounded-3xl"
    >
      <div
        class="flex-row w-full p-5"
      >
        <div
          class=" flex justify-end"
        >
          <span
            class="font-primary text-hunyadi-yellow text-3xl"
          >
            Top 5 sports with the most interest
          </span>
        </div>
        <div
          class="flex justify-evenly w-full py-5"
        />
        <div
          class="flex w-full justify-end pb-5 pr-5 text-white"
        >
          <a
            href="/audience"
          >
            Overall audience page 
            <span
              aria-label="arrow-right"
              class="anticon anticon-arrow-right"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="arrow-right"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M869 487.8L491.2 159.9c-2.9-2.5-6.6-3.9-10.5-3.9h-88.5c-7.4 0-10.8 9.2-5.2 14l350.2 304H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h585.1L386.9 854c-5.6 4.9-2.2 14 5.2 14h91.5c1.9 0 3.8-.7 5.2-2L869 536.2a32.07 32.07 0 000-48.4z"
                />
              </svg>
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
